src/file.rs

```diff
@@ -43,11 +77,66 @@ impl HasPath for PubkyAppFile {
 }
 
 impl Validatable for PubkyAppFile {
-    // TODO: content_type validation.
+    fn sanitize(self) -> Self {
+        let name = self.name.trim().chars().take(MAX_NAME_LENGTH).collect();
+
+        let sanitized_src = self
+            .src
+            .trim()
+            .chars()
+            .take(MAX_SRC_LENGTH)
+            .collect::<String>();
+
+        let src = match Url::parse(&sanitized_src) {
+            Ok(_) => Some(sanitized_src),
+            Err(_) => None, // Invalid src URL, set to None
+        };
+
+        let content_type = self.content_type.trim().to_string();
+
+        Self {
+            name,
+            created_at: self.created_at,
+            src: src.unwrap_or("".to_string()),
+            content_type,
+            size: self.size,
+        }
+    }
+
     fn validate(&self, id: &str) -> Result<(), String> {
         self.validate_id(id)?;
-        // TODO: content_type validation.
-        // TODO: size and other validation.
+
+        // Validate name
+        let name_length = self.name.chars().count();
+
+        if !(MIN_NAME_LENGTH..=MAX_NAME_LENGTH).contains(&name_length) {
+            return Err("Validation Error: Invalid name length".into());
+        }
+
+        // Validate src
+        if self.src.chars().count() == 0 {
+            return Err("Validation Error: Invalid src".into());
+        }
+        if self.src.chars().count() > MAX_SRC_LENGTH {
+            return Err("Validation Error: src exceeds maximum length".into());
+        }
+
+        // validate content type
+        match Mime::from_str(&self.content_type) {
+            Ok(mime) => {
+                if !VALID_MIME_TYPES.contains(&mime.essence_str()) {
+                    return Err("Validation Error: Invalid content type".into());
+                }
+            }
+            Err(_) => {
+                return Err("Validation Error: Invalid content type".into());
+            }
+        }
+
+        // Validate size
+        if self.size <= 0 || self.size > MAX_SIZE {
```

Comment:
Are we taking the client declared `size` in `/pubky.app/file/<object>`  at face value? Is there any way we can actually check the size of the blob?

Reply:
I don't think the user has any incentive to lie about the size of the file.  
Having said that, it looks close to the content-type issue. Our best check would be to make sure it matches the content-length header returned from the homeserver.

Reply:
Is this something we can write into this specs crate? Should be as part of the `/blob` validation that probably has to run as a child validation of the `/file` one ?